canvas_sdk/utils/http.py

```diff
@@ -0,0 +1,55 @@
+import time
+from functools import wraps
+from typing import Any, Callable, TypeVar
+
+import requests
+import statsd
+
+F = TypeVar("F", bound=Callable)
+
+
+class Http:
+    """A helper class for completing HTTP calls with metrics tracking."""
+
+    def __init__(self) -> None:
+        self.session = requests.Session()
```

Comment:
Should we be concerned about different plugins (from different authors) sharing the same session? Would plugins be able to steal cookies from other plugins?

Reply:
i dont think that is possible? within the plugin they will instantiate the helper class, so the session will be limited to just that plugin. e.g.:
```
def compute():
    http = Http()
    http.get('https://www.canvasmedical.com')
```

Reply:
Oh! Duh. Thanks.